### ISSUE TITLE
(Understandable) Misinformation about using SMTP port 465 for notifications in the logs

### ISSUE THREAD

**Issue by georglauterbach** [authelia/authelia#2194]
You may know me from the current OIDC documentation updates. I'm also member of the core maintainers team at [docker-mailserver/docker-mailserver](https://github.com/docker-mailserver/docker-mailserver).

## The Issue at Hand

I was configuring Authelia's notifier settings and stumbled upon this error message

``` INI
level=warning msg="Notifier SMTP client configured to connect to a SMTPS server. It's highly recommended you use a non SMTPS port and STARTTLS instead of SMTPS, as the protocol is long deprecated."
``` 

While this did not bother me in the beginning, I figured I should report this.

## Rationale

First of all, 465 _was_ SMTPS, but isn't anymore. It is now _submission**s**_.

There is **a lot** of misinformation about port 465 on the internet. I'd **highly suggest to read [this section from our mail server documentation](https://docker-mailserver.github.io/docker-mailserver/edge/config/security/understanding-the-ports/)**. It states that port 465 is "a submission port since 2018".

To get the full view, one should read

1. [RFC 8314 Section 7.3](https://datatracker.ietf.org/doc/html/rfc8314#section-7.3) 
2. [RFC 8314 Section 3.3](https://datatracker.ietf.org/doc/html/rfc8314#section-3.3)
3. [RFC 8314 Section 4.1](https://datatracker.ietf.org/doc/html/rfc8314#section-4.1)

TL;DR 465 is recommended over 587 as it communicates the use of TLS more clearly. If configured correctly, there is no security advantage of 465 over 587, but with 465, TLS is _mandatory_, and this is the advantage. The message in the logs is therefore not accurate (outdated ?).

**Comment by james-d-elliott** [comment]
You are correct according to the IANA port registrations, thanks for reporting!

The question in my mind is how many mail servers still run SMTPS and how do we  differentiate between them. It may be worth us just deprecating this log message entirely if we can't differentiate at all. Do you happen to know of any resources listing server implementations that implement SUBMISSIONS vs SMTPS?

Also TLS is also mandatory in our implementation as we drop the connection and throw an error if TLS fails to negotiate unless someone explicitly turns it off (their own choice), meaning there is actually no difference between plain-text -> STARTTLS and regular TLS; this is because if a sever was using STARTTLS to authenticate and someone tried to MITM by not advertising STARTTLS mail delivery would error and terminate before it initiated the RCPT TO. The notice was specifically in regards to the SMTPS deprecation itself.

**Comment by georglauterbach** [comment]
> You are correct according to the IANA port registrations, thanks for reporting!

Glad I can help.

> Do you happen to know of any resources listing server implementations that implement SUBMISSIONS vs SMTPS?

I think that there is no such list.

> The question in my mind is how many mail servers still run SMTPS (probably not many many would think since it was long deprecated however the number of mail servers supporting SMTPS post-2000/pre-2018 is huge) and how do we differentiate between them. It may be worth us just deprecating this log message entirely if we can't differentiate at all.

I think that would be good. Because as is, IANA / IETF _actually recommend_ using 465 over 587 and try to get people to transition there. This log message discourages users to do so. It would be unwise to leave this message there just because the number of servers is large. The right way would be to inform users through the documentation if you ask me :)

> Also TLS is also mandatory in our implementation as we drop the connection and throw an error if TLS fails to negotiate unless someone explicitly turns it off (their own choice), meaning there is actually no difference between plain-text -> STARTTLS and regular TLS; this is because if a sever was using STARTTLS to authenticate and someone tried to MITM by not advertising STARTTLS mail delivery would error and terminate before it initiated the RCPT TO.

Yes, like I said. No real benefit in terms of security _if configured correctly_, but only if :)

> The notice was specifically in regards to the SMTPS deprecation itself.

Understandable. But this log message should go. I can try to come up with a PR, though I have never written a single line of Go. There is always a first I guess :D I will get to this in a few hours.
